pandas/core/dtypes/cast.py

```diff
@@ -312,12 +314,11 @@ def maybe_cast_result_dtype(dtype: DtypeObj, how: str) -> DtypeObj:
     DtypeObj
         The desired dtype of the result.
     """
-    d = {
-        (np.dtype(np.bool), "add"): np.dtype(np.int64),
-        (np.dtype(np.bool), "cumsum"): np.dtype(np.int64),
-        (np.dtype(np.bool), "sum"): np.dtype(np.int64),
-    }
-    return d.get((dtype, how), dtype)
+    if how in ["add", "cumsum", "sum"] and (dtype == np.dtype(np.bool)):
+        return np.dtype(np.int64)
+    if how in ["add", "cumsum", "sum"] and isinstance(dtype, pd.BooleanDtype):
```

Comment:
dont use ``pd`` import *anywhere*, rather prefer to import directly.

Reply:
I'm hitting some circular import issues when I try to import these directly, is it okay to import from within the function itself?